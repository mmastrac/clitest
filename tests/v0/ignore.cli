#!/usr/bin/env clitest --v0

ignore {
    ! IGNORED%{GREEDYDATA}
}

$ echo "IGNORED something\na"
! a

$ echo "a\nIGNORED something\nb"
! a
! b

$ echo "a\nXIGNORED something\nb\nXIGNORED something else\nc"
! a
sequence {
    ignore {
        ! XIGNORED%{GREEDYDATA}
    }
    ! b
    ! c
}

$ echo "LOG first\na\nLOG something\nb\nLOG something else\nc"
%EXPECT_FAILURE
! a
sequence {
    ignore {
        ! LOG %{GREEDYDATA}
    }
    ! b
    ! c
}
