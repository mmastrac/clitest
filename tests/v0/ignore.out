Running <root>/ignore.cli ...

printf "IGNORED something\na\n"
---
IGNORED something
a
---
✅ OK

printf "a\nIGNORED something\nb\n"
---
a
IGNORED something
b
---
✅ OK

printf "a\nXIGNORED something\nb\nXIGNORED something else\nc\n"
---
a
XIGNORED something
b
XIGNORED something else
c
---
✅ OK

printf "LOG first\na\nLOG something\nb\nLOG something else\nc\n"
---
LOG first
a
LOG something
b
LOG something else
c
---
✅ OK

<root>/ignore.cli PASSED
1 test(s) passed
